src/others/helpers/decode-aid-request.ts

```diff
@@ -71,7 +84,9 @@ export const decodeAidRequestGroupedByCategoryWithTotal = (dictionary: Dictionar
 
 const decodeLocation = (locations: Location[], cityId: ID): DecodedLocation => {
   const location = locations.find((location) => String(location.id) === String(cityId));
-  return omit(location, "id");
+  if (!location) throw new Error(`Locatin: ${location} is not defined`);
```

Comment:
why do we still have this decodeLocation function that does a linear scan? why not switch the users to use DictionaryContext?

Reply:
According to TODO (line 5 - // TODO: Delete this file once map-utils start using raw AidRequest only), this whole file will be deleted after migrating the map into using context. The same applies to the next comment.

Also nit from me: "Error(`Location: ${location} is not defined`)"

Reply:
Yes, we'll need to refactor map component and related helpers so that it relies on shared data models: 
- Supply
- Location
- AidRequest

And the necessary transformation / translations are performed lower in the components tree, where they are needed. 

All this "decoding" was a bad architectural decision of mine, and I take full responsibility for it :)